### DISCUSSION TITLE
Allow readonly as disabled

### DISCUSSION THREAD

**Discussion by eusonlito** [orchidjs/tom-select#304] [Ideas]
The `disabled` and` readonly` properties are similar, but a form with a `disabled` input/select will not send the field to the backend, but with` readonly` it will.

In many cases the field needs to exist as it is a backend independent validation.

I think adding `readonly` feature like it is now` disabled` would be a good idea :)

Thanks!

**Comment by nallwhy** [comment]
It's true!

**Comment by ttskch** [comment]
The `readonly` attribute is [not defined for the `select` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly). Assigning an undefined attribute should be avoided as much as possible.

Tom Select provides the [`lock()` method](https://tom-select.js.org/examples/lock/) to disable operations on the control, so it is appropriate to use this.

Alternatively, you can simply achieve `readonly` feature by defining the following CSS and assigning the `data-readonly` attribute to the `select` element.

```css
select[data-readonly] + .ts-wrapper {
  background-color: #fff;
  opacity: 0.5;
  pointer-events: none;
}
```

Regards.
